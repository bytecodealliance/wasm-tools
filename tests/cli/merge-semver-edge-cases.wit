// RUN[new-import-new-dep]: component embed --dummy --world new-dep-a % | \
//                          component embed         --world new-dep-b % | \
//                          component wit
//
// RUN[reorder-new-dep]: component embed --dummy --world reorder-dep-a % | \
//                       component embed         --world reorder-dep-b % | \
//                       component wit

package foo:bar;

world new-dep-a {
  import a:b/foo@0.1.0;
}

world new-dep-b {
  import a:b/foo@0.1.1;
}

world reorder-dep-a {
  import a:b/add-dep-on-prev-foo@0.1.0;
  import a:b/foo@0.1.0;
}

world reorder-dep-b {
  import a:b/add-dep-on-prev-foo@0.1.1;
}

package a:b@0.1.0 {
  interface foo {
    type t = u32;
  }

  interface add-dep-on-prev-foo {}
}

package a:b@0.1.1 {
  interface foo {
    use bar.{t};
  }

  interface bar {
    type t = u32;
  }

  interface add-dep-on-prev-foo {
    use a:b/foo@0.1.0.{t};
  }
}
