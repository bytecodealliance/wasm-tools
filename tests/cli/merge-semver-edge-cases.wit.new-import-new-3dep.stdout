package root:root;

world root {
  import a:b/bar@0.1.1;
  import a:b/foo@0.1.1;
}
package a:b@0.1.1 {
  interface bar {
    type t = u32;
  }
  interface foo {
    use bar.{t};
  }
}


package foo:bar {
}


package a:b@0.1.0 {
  interface foo {
  }
}
